src/services/auth.ts

```diff
@@ -175,34 +186,40 @@ class AuthService {
         oidcUser = null;
       }
     }
-    let user: User | null = null;
     if (oidcUser) {
-      const { name, email, sub } = oidcUser.profile;
-      if (name && email && sub) {
+      const { sub, name, email } = oidcUser.profile;
+      if (sub && name && email) {
         const expired = oidcUser.expired ?? true;
         let fullName = name;
         user = {
           expired,
-          name,
+          name: name,
           fullName: name,
+          loginState: LoginState.Identified,
           email,
           extId: sub,
         };
         try {
-          const response = await fetchJson(new URL(sub, USERS_URL));
+          let tokenHeader: Record<string, string> = { "X-Authorization": "Bearer " + oidcUser.access_token }
+          const response = await fetchJson(new URL(sub, USERS_URL), "GET", null, tokenHeader);
           if (response.status === 200) {
             const userData = await response.json();
-            const { id, status, title } = userData;
+            const { id, title } = userData;
             if (title) fullName = `${title} ${fullName}`;
+            user.loginState = name !== userData.name || email !== userData.email ? LoginState.NeedsReregistration : LoginState.Registered;
             user = {
               ...user,
               id,
-              status,
               title,
               fullName,
-              changed: name !== userData.name || email !== userData.email,
             };
-          } else if (response.status !== 404) {
+          } else if (response.status === 404) {
+            user = {
```

Comment:
Maybe just `user.loginState = LoginState.NeedsRegistration` as you're doing above. Easier to read.